core/ui/public/i18n/en/translation.json

```diff
@@ -207,6 +207,8 @@
         "the_join_code_can_not_be_decoded": "Join code cannot be decoded",
         "the_join_code_is_not_correctly_encoded": "Join code is not correctly encoded",
         "the_join_code_cannot_be_parsed": "Join code cannot be parsed",
+        "core_version_mismatch": "To join this cluster core version {value} is required",
```

Comment:
I'd suggest a solution here because is a blocking condition.
Maybe something like:
```
To join this cluster core version {value} is required. Make sure to update the cluster leader or reinstall the worker with the latest available image.
```